internal/fleet/integration_policy/secrets.go

```diff
@@ -0,0 +1,165 @@
+package integration_policy
+
+import (
+	"context"
+	"encoding/json"
+
+	fleetapi "github.com/elastic/terraform-provider-elasticstack/generated/fleet"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+// The secret store is a map of policy secret reference IDs to the
+// original value at time of creation. By replacing the ref when
+// marshaling the state back to Terraform, we can prevent resource
+// drift.
+type secretStore map[string]any
+
+// newSecretStore creates a new secretStore from the resource privateData.
+func newSecretStore(ctx context.Context, private privateData) (store secretStore, diags diag.Diagnostics) {
+	bytes, diags := private.GetKey(ctx, "secrets")
+	if diags != nil {
+		return
+	}
+	if bytes == nil {
+		bytes = []byte("{}")
+	}
```

Comment:
```suggestion
	if bytes == nil {
		return
	}
```

Can we just return directly here instead?

Reply:
We can init an empty map and return, sure.